src/progress/bar.rs

```diff
@@ -452,17 +454,29 @@ impl Bar {
     }
 
     pub(crate) fn fmt_rate(&self) -> String {
-        format!(
-            "{}{}/s",
-            if self.counter == 0 {
-                "?".to_owned()
-            } else if self.unit_scale {
-                format::format_sizeof(self.rate() as f64, self.unit_divisor as f64)
-            } else {
-                format!("{:.2}", self.rate())
-            },
-            self.unit
-        )
+        if self.counter == 0 {
+            format!("?{}/s", self.unit)
+        } else if self.rate() < 1. {
```

Comment:
This feature is good but first refactor code a bit. `self.rate()` is called here and again inside the block. Create a variable to store temporary rate value.